env/io_posix.cc

```diff
@@ -875,9 +875,12 @@ IOStatus PosixRandomAccessFile::ReadAsync(
     }
   }
 
-  // Init failed, platform doesn't support io_uring.
+  // Init failed, platform doesn't support io_uring. Fallback to Read API.
   if (iu == nullptr) {
-    return IOStatus::NotSupported("ReadAsync");
+    req.status =
+        Read(req.offset, req.len, opts, &(req.result), req.scratch, dbg);
+    cb(req, cb_arg);
+    return IOStatus::OK();
```

Comment:
Sorry to go back and forth on this, but I'm thinking it would be better to stick to returning NotSupported, since there is a performance downside to silently doing sync read instead of async. IOW, MultiGet/scan with async_io and silently switching to sync reads will perform worse than MultiGet/scan without async_io flag. So its better to give the user some indication until we figure out a way to avoid the performance penalty.

I think its good to fail if IO uring is not supported even for db_stress, as we will know exactly what's being tested.